### PR TITLE
Try using iframe for loading assembler hub behind the scenes

### PR DESCRIPTION
**Pull request by ilyasfoo**
### Changes proposed in this Pull Request:

Attempt using iframe for temporary solution on https://github.com/woocommerce/woocommerce/issues/40793

Closes https://github.com/woocommerce/woocommerce/pull/40936

### How to test the changes in this Pull Request:

1. Make sure to enable `customize-store` feature flag
1. Go to `WooCommerce -> Home` and click `Customize Store` task
1. Go through the AI wizard and observe the assembler hub showing up relatively quickly after the loading screen
1. Please test around navigating links within the iframe, it should behave like it's the original window
2. Please test with other browsers

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/design-with-ai/actions.ts [hunk 1]
```diff
@@ -276,12 +276,43 @@ const recordTracksStepCompleted = (
...
+	iframe.src = assemblerUrl; // Replace with the URL of the page you want to load
+
+	iframe.onload = () => {
+		// Hide loading UI
+		attachIframeListeners( iframe );
+		// Make iframe fullscreen
+		setTimeout( () => {
+			iframe.style.display = 'block';
+			iframe.classList.add( 'cys-fullscreen-iframe' );
+		}, 6000 );
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/customize-store/design-with-ai/actions.ts]: 6 seconds lead time for loading to happen in background

**Review by moon0326 — COMMENTED** [review]
Great work on this @ilyasfoo!

It worked really well. In my case, I did not even see the loading screen.

I think we can make a a few tweaks to remove hard-coded 6000 seconds logic.

What do you think about starting the iframe when we hit [pending](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L405) state and make the iframe visible on [onDone](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L425) state? 

I usually see the loading screen for 2~3 seconds and 6 seconds might be unnecessary for many users. Preloading the iframe while requesting the API endpoints should be sufficient in my opinion.

**Comment by ilyasfoo** [comment]
> What do you think about starting the iframe when we hit [pending](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L405) state and make the iframe visible on [onDone](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L425) state?

> I usually see the loading screen for 2~3 seconds and 6 seconds might be unnecessary for many users. Preloading the iframe while requesting the API endpoints should be sufficient in my opinion.

@moon0326 Thanks for testing! I don't think that's advisable, as the step can still fail and loading the iframe is resource-intensive. I think reducing the timeout by a couple seconds may be a good compromise?

**Comment by moon0326** [comment]
> > What do you think about starting the iframe when we hit [pending](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L405) state and make the iframe visible on [onDone](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/design-with-ai/state-machine.tsx#L425) state?
> 
> > I usually see the loading screen for 2~3 seconds and 6 seconds might be unnecessary for many users. Preloading the iframe while requesting the API endpoints should be sufficient in my opinion.
> 
> @moon0326 Thanks for testing! I don't think that's advisable, as the step can still fail and loading the iframe is resource-intensive. I think reducing the timeout by a couple seconds may be a good compromise?

Reducing the timeout by a couple seconds sounds good enough 👍 

My thought was that there are more successes than failures. Even in case of failure, the browser caches assets. These cached assets can be loaded from memory on the second attempt.

File: plugins/woocommerce-admin/client/customize-store/design-with-ai/actions.ts (hunk 1 above)
Comment (ilyasfoo) [inline plugins/woocommerce-admin/client/customize-store/design-with-ai/actions.ts]:
```suggestion
		}, 4000 );
```

**Review by moon0326 — APPROVED** [review]
LGTM and tested well 👍 🚀

**Review by adrianduffell — APPROVED** [review]
It's testing super nice on my connection -- thank you @ilyasfoo ! 🚀

**Pull request merged**